Bottom: 8b4ffe16e809c1339e7cd2be5524e63648b2a93a
Top:    a92e31761300471a59792ce8a5b13400b6a4c1a7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-23 20:56:22 +0000

find.c: Fix inverted logic in LookupConnection()

When AndRats is true, we may consider starting with a ratline object,
otherwise we do no. I got this backwards by mistake.


---

diff --git a/src/find.c b/src/find.c
index 5f1cfac..f56c947 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3048,7 +3048,7 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
   if (type == NO_TYPE)
     {
       type = SearchObjectByLocation (
-        LOOKUP_MORE & ~(AndRats ? RATLINE_TYPE : 0),
+        LOOKUP_MORE & ~(AndRats ? 0 : RATLINE_TYPE),
         &ptr1, &ptr2, &ptr3, X, Y, Range);
       if (type == NO_TYPE)
         return;
